### PR TITLE
UIDATIMP-166: Add Search box for 3rd pane of Match Profiles

### PR DESCRIPTION
**Pull request by SergiyVSergiyenko**
## Purpose
Add Search box for 3rd pane of Match Profiles according to requirements in the [story](https://issues.folio.org/browse/UIDATIMP-166).

## Screenshots
Success_1
Success_2
Empty

### CODE REVIEW

File: src/utils/trimSearchTerm.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+/**
+ * Remove '*' symbols from both sides of a string
+ *
+ * @param {string} a
+ * @return {string} Trimmed string
+ */
+export const trimSearchTerm = a => a.replace(/^(\*+)/, '').replace(/(\*+)$/, '');
```
Comment (OleksandrYehorovEPAM) [inline src/utils/trimSearchTerm.js]: I would give a more meaningful name to the `a` parameter. 
```suggestion
export const trimSearchTerm = searchTerm => searchTerm.replace(/^(\*+)/, '').replace(/(\*+)$/, '');
```

File: src/settings/MatchProfiles/MatchProfiles.js [hunk 2]
```diff
@@ -133,7 +146,7 @@ export class MatchProfiles extends Component {
     } = this.props;
 
     const urlQuery = queryString.parse(search);
-    const searchTerm = (urlQuery.query || '').trim();
+    const searchTerm = trimSearchTerm((urlQuery.query || '').trim());
```
Comment (OleksandrYehorovEPAM) [inline src/settings/MatchProfiles/MatchProfiles.js]: Consider moving `.trim()` call in `trimSearchTerm` function as it looks weird to trim 2 times

Comment (VictorSoroka1) [reply src/settings/MatchProfiles/MatchProfiles.js]: I would insist to move to `trimSearchTerm`.

File: src/utils/trimSearchTerm.js (hunk 1 above)
Comment (VictorSoroka1) [reply src/utils/trimSearchTerm.js]: And do not forget to update jsDoc after parameter rename.

File: test/bigtest/interactors/match-profiles.js [hunk 3]
```diff
@@ -1,16 +1,22 @@
...
 
 import { CheckboxInteractor } from './checkbox-interactor';
 
 @interactor class MatchProfilesInteractor {
   list = new MultiColumnListInteractor('#match-profiles-list');
   selectAllCheckBox = new CheckboxInteractor('[data-test-select-all-checkbox]');
   checkBoxes = collection('[data-test-select-item]', CheckboxInteractor);
+  searchField = new TextFieldInteractor('#input-match-profiles-search');
+  searchSubmitButton = new ButtonInteractor('[data-test-search-and-sort-submit]');
+  searchSubmitButtonDisabled = property('[data-test-search-and-sort-submit]', 'disabled');
```
Comment (VictorSoroka1) [inline test/bigtest/interactors/match-profiles.js]: I do not see much sense in that additional prop as we tend to use `searchSubmitButton.text`. So please proceed with `searchSubmitButton.disabled` use.

File: src/settings/MatchProfiles/MatchProfiles.js [hunk 4]
```diff
@@ -14,6 +14,8 @@ import {
 import { makeQueryFunction } from '@folio/stripes/smart-components';
 import { Checkbox } from '@folio/stripes/components';
 
+import { trimSearchTerm } from '../../utils';
+
```
Comment (VictorSoroka1) [inline src/settings/MatchProfiles/MatchProfiles.js]: Please remove this empty line. We do not group utils imports.

File: src/utils/createSearchTerm.js [hunk 5]
```diff
@@ -0,0 +1,9 @@
+/**
+ * Remove whitespaces and '*' symbols from both sides of a string
+ *
+ * @param {object} searchInfo
+ * @return {string} Trimmed string
+ */
+export const createSearchTerm = urlQuery => {
```
Comment (OleksandrYehorovEPAM) [inline src/utils/createSearchTerm.js]: Let's have this function to take a string as an argument

File: src/settings/MatchProfiles/MatchProfiles.js [hunk 6]
```diff
@@ -24,6 +25,17 @@ import sharedCss from '../../shared.css';
 const INITIAL_RESULT_COUNT = 30;
 const RESULT_COUNT_INCREMENT = 30;
 
+const findAllQueryParameter = 'cql.allRecords=1';
+const matchProfileQueryTemplate = `(
```
Comment (VictorSoroka1) [inline src/settings/MatchProfiles/MatchProfiles.js]: Name it just `queryTemplate` per consistency.

**Comment by OleksandrYehorovEPAM** [comment]
Please add a note to changelog.md

File: src/settings/MatchProfiles/MatchProfiles.js [hunk 7]
```diff
@@ -133,7 +145,7 @@ export class MatchProfiles extends Component {
     } = this.props;
 
     const urlQuery = queryString.parse(search);
-    const searchTerm = (urlQuery.query || '').trim();
+    const searchTerm = trimSearchTerm(urlQuery.query);
```
Comment (VictorSoroka1) [inline src/settings/MatchProfiles/MatchProfiles.js]: Please apply the same for the job profiles and file extensions.

**Pull request merged**